hollow/src/main/java/com/netflix/hollow/core/write/objectmapper/HollowSetTypeMapper.java

```diff
@@ -132,6 +136,18 @@ protected Object parseFlatRecord(HollowSchema recordSchema, FlatRecordReader rea
         return collection;
     }
 
+    @Override
+    protected Object parseFlatRecordTraversalNode(FlatRecordTraversalNode node) {
+        FlatRecordTraversalSetNode setNode = (FlatRecordTraversalSetNode) node;
+        Set<Object> collection = new HashSet<>();
+        Iterator<FlatRecordTraversalNode> elements = setNode.iterator();
+        while (elements.hasNext()) {
```

Comment:
nit: can simplify this since SetNode/ListNode are iterables
```suggestion
       for (FlatRecordTraversalNode element : setNode) {
```